### PR TITLE
Elixir: Improve generated API options docs

### PR DESCRIPTION
**Pull request by halostatue**
@mrmstn

While working with the Magento 2.4 API definition to generate an Elixir client, there were several issues found. The easiest to resolve is a documentation naming issue. (The others will be raised for discussion.)

This is known to be an issue with query parameters, but may not be limited to *just* query parameters.

In the event that a parameter is a mix of arrays and objects, the template in the documentation for the API function was expanding out the parameter name such that `foo[0]` would be turned into `:foo_left_square_bracket0_right_square_bracket`, but the actual parameter name was being left as `:foo[0]`, which meant that the documentation did not reflect the actual parameter.

> Note: there are issues with the way that query parameters are put
> together in this sense, which is going to require substantially more
> work to resolve as well as discussion on how these options should be
> implemented, as what the Magento API requires may not be what is
> expected by a different server, and the nature of the input parameters
> is itself incorrect.

~As a small drive-by fix, changed `run-in-docker.sh` to use `set -euo pipefail` instead of `set -exo pipefail`. The script does not need tracing turned on but should check for previously unreferenced variables. This makes the output of `run-in-docker.sh` slightly nicer.~

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: run-in-docker.sh
Comment (mrmstn) [inline run-in-docker.sh:2]: Although it looks like a small change, this would need a review from the core team, since it's not elixir specific

Could you separate this one out of the PR?

**Review by mrmstn — CHANGES_REQUESTED** [review]
Hi @halostatue, thank you very much for your PR and the resulting documentation fix 😁👌🏻 

the elixir specific changes look good to me, the drive-by fix is out of scope but it would be great if you could address this issue through a separate PR (or Issue for discussion)

File: run-in-docker.sh
Comment (wing328) [reply run-in-docker.sh:2]: @mrmstn good catch.

@halostatue What problem is this change going to solve?

Comment (halostatue) [reply run-in-docker.sh:2]: I can separate it out pretty easily. It’s a pair of minor changes, each of which could be separated.

1. `set -x` turns on tracing. With this, if you have:

   ```sh
   #!/usr/bin/env bash
   set -x
   echo Hello
   ```

   The output would be:

   ```console
   $ ./hello.sh
   + echo Hello
   Hello
   ```

   Removing the `x` from that cleans up the output.

2. `set -u` turns on errors for unset variables. Usually more useful on larger scripts, this would prevent the following from running (usually in error):

   ```sh
   #!/usr/bin/env bash
   echo $hello
   ```

   It’s fairly useful for catching variable name typos. (`echo $color` when your variable is `$colour`.)

I’m seeing a separate issue that should be addressed in this and other shell scripts, so I will pull it into its own PR anyway (hardcoding `#!/bin/bash` is an antipattern; the version of bash at `/bin`, if it exists, may not be the best version of bash to use on the system).

**Review by mrmstn — APPROVED** [review]

**Comment by wing328** [comment]
For the CircleCI failure,  we can ignore it.

I ran the tests locally and the result is good:
```
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  earmark_parser 1.4.26
  ex_doc 0.28.5
  makeup 1.1.0
  makeup_elixir 0.16.0
  makeup_erlang 0.1.1
  mime 2.0.3
  nimble_parsec 1.2.3
  poison 3.1.0
  tesla 1.4.4
All dependencies are up to date
[INFO] 
[INFO] --- exec-maven-plugin:1.2.1:exec (test) @ ElixirPetstoreClientTests ---
...

Finished in 0.6 seconds (0.00s async, 0.6s sync)
3 tests, 0 failures

Randomized with seed 561935
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  6.947 s
[INFO] Finished at: 2022-09-26T16:34:08+08:00
[INFO] ------------------------------------------------------------------------
```

Please feel free to merge.

**Pull request merged**
